docs/DeveloperGuide.md

```diff
@@ -2,12 +2,112 @@
 
+![IncomeManager_seqdiagram.drawio.png](..%2Fimages%2FIncomeManager_seqdiagram.drawio.png)
+
+### ExpenseManager
+
+The ExpenseManager is facilitated by `ExpenseParser`,  `ExpenseManager`, `ExpenseList` and `Ui`
+
+1. The user inputs the command to add expense.
+2. This command is used to instantiate a ExpenseManager object which is executed.
+3. During execution, the command is passed to ExpenseParser which extracts and parses the input fields.
+4. Input field information is passed to the relevant Expense constructor which returns a Expense object.
+5. Expense object is returned to ExpenseManager from ExpenseParser.
+6. Expense obejct is returned to Duke from ExpenseManager.
+
+![expenseManager_sequence_diagram.png](/Users/apple/Desktop/CS2113/tp/images/expenseManager_sequence_diagram.png)
+
+### Find Command
+The FindCommand is facilitated by `FindCommand`, `FindParser`, `Ui`,`ExpenseList` and `IncomeList`.
+
+1. The user will first input the command to find the expense or income.
+```
+find /t <type> /cat [category] /de [description] /date [date]
+```
+2. The inputs will then be passed to `FindParser` to get the necessary fields such as type, category, description and date.
+3. The FindCommand is then instantiated with the necessary fields.
+4. Users can choose to search through the expense list or income list.
+5. If users searches through the income list, either the description or date will be used to search through the income list.
+6. If users searches through the expense list, either the category or date will be used to search through the expense list.
+7. The execute method will then be called to search through the list and return the list of expenses or incomes that matches the search criteria.
+
+![FindCommand_SequenceDiagram.png](..%2Fimages%2FFindCommand_SequenceDiagram.png)
+
+### GetFromTxt Command
+The GetFromTxt Command is facilitated by `IOException`,`Scanner`, `KaChinnnnngException`, `IncomeList` and `ExpenseList`
+1. The user start the program and the command will start automatically with a default path
+2. Once the command executed, it will try to create an txt file on the default path
+3. If file failed to create, throw an `IOException`
+4. Catch the `IOExceotion` and print error messages
+5. Create a `Scanner` Object s to get content from txt file
+6. If amount in txt file exceed limit or incorrect, throw `KaChinnnningException`
+7. Read the content on the file and add corresponding object to incomes or expenses list
+
+![FindCommand_SequenceDiagram.png](..%2Fimages%2FGetFromTxt.png)
+
+### ExchangeRateManager
+The Sequence Diagram below shows how the components interact with each other for the scenario 
+where the user issues the command `update exchange rate USD 0.8`.
+
+![UpdateExchangeRate_SequenceDiagram.png](..%2Fimages%2FUpdateExchangeRate_SequenceDiagram.png)
+
+This is facilitated by `Duke`, `UpdateExchangeRateCommand`, `ExchangeRateManager`'and `ExchangeRateFileHandler`.
```

Comment:
Could the ExchangeRateManager be more descriptive to aid in understanding using the sequence diagram?